### ISSUE TITLE
when run evaluate_model.lua, I get an error.

### ISSUE THREAD

**Issue by XHUJOY** [jcjohnson/densecap#29]
xhu@linux:~/Projects/torch/examples/mydensecap$ th evaluate_model.lua 
Loaded model	
Using dtype "torch.CudaTensor"	
DataLoader loading json file: 	data/VG-regions-dicts.json	
DataLoader loading h5 file: 	data/VG-regions.h5	
reading box_to_img	
reading boxes	
reading image_heights	
reading image_widths	
reading img_to_first_box	
reading img_to_last_box	
reading labels	
reading lengths	
reading original_heights	
reading original_widths	
reading split	
assigned 77398/5000/5000 images to train/val/test.	
initialized DataLoader:	
#images: 87398, #regions: 4158176, sequence max length: 15	
using split 	1	
Processed image 59.jpg (1 / 5000) of split 1, detected 103 regions	
Processed image 67.jpg (2 / 5000) of split 1, detected 92 regions	
Processed image 78.jpg (3 / 5000) of split 1, detected 83 regions	
/home/xhu/Projects/torch/install/bin/luajit: ./densecap/LanguageModel.lua:98: attempt to concatenate a nil value
stack traceback:
	./densecap/LanguageModel.lua:98: in function 'decodeSequence'
	./eval/eval_utils.lua:63: in function 'eval_split'
	evaluate_model.lua:63: in main chunk
	[C]: in function 'dofile'
	...ects/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00405ea0

when i run evaluate_model.lua , I get an error like this. I have no idea about this.

**Comment by XHUJOY** [comment]
evaluating val dataset during trainning is OK.

xhu@linux:~/Projects/torch/examples/mydensecap$ th train.lua 
{
  finetune_cnn_after : -1
  weight_decay : 1e-06
  optim_beta2 : 0.999
  val_images_use : 1000
  timing : false
  optim_epsilon : 1e-08
  test_num_proposals : 1000
  input_encoding_size : 512
  losses_log_every : 10
  id : ""
  gpu : 0
  drop_prob : 0.5
  data_json : "data/VG-regions-dicts.json"
  clip_final_boxes : 1
  sampler_batch_size : 256
  rnn_size : 512
  test_rpn_nms_thresh : 0.7
  progress_dump_every : 100
  optim_beta1 : 0.9
  end_objectness_weight : 0.1
  seed : 123
  test_final_nms_thresh : 0.3
  eval_first_iteration : 1
  captioning_weight : 1
  checkpoint_start_from : ""
  sampler_high_thresh : 0.7
  mid_objectness_weight : 0.1
  data_h5 : "data/VG-regions.h5"
  train_remove_outbounds_boxes : 1
  sampler_low_thresh : 0.3
  debug_max_train_images : -1
  learning_rate : 1e-05
  rpn_hidden_dim : 512
  max_iters : -1
  checkpoint_path : "checkpoint.t7"
  save_checkpoint_every : 10000
  end_box_reg_weight : 0.1
  proposal_regions_h5 : ""
  backend : "cudnn"
  mid_box_reg_weight : 0.05
  box_reg_decay : 5e-05
}
DataLoader loading json file: 	data/VG-regions-dicts.json	
DataLoader loading h5 file: 	data/VG-regions.h5	
reading box_to_img	
reading boxes	
reading image_heights	
reading image_widths	
reading img_to_first_box	
reading img_to_last_box	
reading labels	
reading lengths	
reading original_heights	
reading original_widths	
reading split	
assigned 77398/5000/5000 images to train/val/test.	
initialized DataLoader:	
#images: 87398, #regions: 4158176, sequence max length: 15	
initializing a DenseCap model from scratch...	
loading network weights from .. VGG_ILSVRC_16_layers.caffemodel	
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:537] Reading dangerously large protocol message.  If the message turns out to be larger than 1073741824 bytes, parsing will be halted for security reasons.  To increase the limit (or to disable these warnings), see CodedInputStream::SetTotalBytesLimit() in google/protobuf/io/coded_stream.h.
[libprotobuf WARNING google/protobuf/io/coded_stream.cc:78] The total number of bytes read was 553432081
Successfully loaded /home/xhu/Projects/torch/examples/mydensecap/data/models/vgg-16/VGG_ILSVRC_16_layers.caffemodel
conv1_1: 64 3 3 3
conv1_2: 64 64 3 3
conv2_1: 128 64 3 3
conv2_2: 128 128 3 3
conv3_1: 256 128 3 3
conv3_2: 256 256 3 3
conv3_3: 256 256 3 3
conv4_1: 512 256 3 3
conv4_2: 512 512 3 3
conv4_3: 512 512 3 3
conv5_1: 512 512 3 3
conv5_2: 512 512 3 3
conv5_3: 512 512 3 3
fc6: 1 1 25088 4096
fc7: 1 1 4096 4096
fc8: 1 1 4096 1000
total number of parameters in net: 	135854800	
total number of parameters in CNN: 	14454528	
iter 0: mid_box_reg_loss: 0.001, captioning_loss: 55.616, end_objectness_loss: 0.088, mid_objectness_loss: 0.150, end_box_reg_loss: 0.003,  [total: 111.714]	
using split 	1	
Processed image 59.jpg (1 / 1000) of split 1, detected 185 regions	
Processed image 67.jpg (2 / 1000) of split 1, detected 189 regions	
Processed image 78.jpg (3 / 1000) of split 1, detected 214 regions	
Processed image 122.jpg (4 / 1000) of split 1, detected 194 regions	
Processed image 185.jpg (5 / 1000) of split 1, detected 180 regions	
Processed image 195.jpg (6 / 1000) of split 1, detected 169 regions	
Processed image 206.jpg (7 / 1000) of split 1, detected 185 regions	
Processed image 262.jpg (8 / 1000) of split 1, detected 202 regions	
Processed image 314.jpg (9 / 1000) of split 1, detected 194 regions	
Processed image 329.jpg (10 / 1000) of split 1, detected 186 regions
